airflow/utils/task_group.py

```diff
@@ -176,6 +177,10 @@ def _check_for_group_id_collisions(self, add_suffix_on_collision: bool):
             else:
                 self._group_id = f"{base}__{suffixes[-1] + 1}"
 
+    def _update_default_args(self, parent_group: TaskGroup):
+        if parent_group.default_args:
+            self.default_args = {**self.default_args, **parent_group.default_args}
```

Comment:
@hussein-awala I was just wondering something, does it mean that parent task group will override default args of a nested task group that also provides `default_args` ? I am not at home so couldn't try it myself, maybe I'm missing something.

For instance:
```python
        @task_group(
            group_id="task_group",
            default_args={
                "owner": "y",
                "execution_timeout": timedelta(seconds=10),
            },
        )
        def tg():
            @task_group(
                group_id="nested_task_group",
                default_args={
                    "owner": "y",
                    "execution_timeout": timedelta(seconds=10),
                },
            )
            def nested_tg():
```

Is it what we want, or should we reverse the order ?
```python
self.default_args = {**parent_group.default_args, **self.default_args}
```

Reply:
Indeed, nice catch :+1: 
can your check #31776?